### PR TITLE
[fakeintake] Re-enable client integration tests

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Re-enabling client integration tests, will investigate why they were disabled, possibly a merge conflict. This PR also improves tests by targeting public client APIs

### Motivation

fakeintake client integration tests should run on pipelines

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]

File: test/fakeintake/client/client_integration_test.go [hunk 1]
```diff
@@ -67,74 +42,66 @@ func TestIntegrationClient(t *testing.T) {
...
-		assert.Equal(t, 1, len(payloads))
-		assert.Equal(t, "totoro|5|tag:valid,owner:pducolin", string(payloads[0].Data))
-		assert.Equal(t, "text/plain", payloads[0].Encoding)
-		assert.Equal(t, mockClock.Now().UTC(), payloads[0].Timestamp)
+		stats, err := client.RouteStats()
+		require.NoError(t, err, "Error getting payloads")
+		expectedStats := map[string]int{
+			"/foo/bar": 1,
+		}
+		assert.Equal(t, expectedStats, stats)
```
Comment (KevinFairise2) [inline test/fakeintake/client/client_integration_test.go:51]: **❓ question:** ‏

IIUC we now test that we have the expected number of payload on the endpoint. But we do not test for the content anymore. I think the test is less precise now. But maybe we do not really care and this is enough. WDYT?

**Review by KevinFairise2 — APPROVED** [review]

File: test/fakeintake/client/client_integration_test.go (hunk 1 above)
Comment (pducolin) [reply test/fakeintake/client/client_integration_test.go:51]: Indeed content is tested on the server test side, here we test the integration, but I should add an integration test with a test fixture payloads that the client can read

**Pull request merged**